pkg/controller/master/setting/ntp_server.go

```diff
@@ -0,0 +1,91 @@
+package setting
+
+import (
+	"reflect"
+	"strings"
+
+	nodev1 "github.com/harvester/node-manager/pkg/apis/node.harvesterhci.io/v1beta1"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	harvesterv1 "github.com/harvester/harvester/pkg/apis/harvesterhci.io/v1beta1"
+	"github.com/harvester/harvester/pkg/util"
+)
+
+const (
+	nodeConfigNS = "harvester-system"
+)
+
+func (h *Handler) syncNTPServer(setting *harvesterv1.Setting) error {
+	logrus.Infof("Processing setting %s, value: %s", setting.Name, setting.Value)
+
+	ntpServers := strings.Replace(setting.Value, ",", " ", -1)
+	if ntpServers == "" {
+		logrus.Infof("NTP server is empty, do no-op here.")
+		return nil
+	}
+
+	nodes, err := h.nodeClient.List(metav1.ListOptions{})
+	if err != nil {
+		logrus.Errorf("List all node failure: %v", err)
+		return err
+	}
+
+	for _, node := range nodes.Items {
+		nodeName := node.Name
+		nodeConfig, err := h.nodeConfigs.Get(nodeConfigNS, nodeName, metav1.GetOptions{})
+		if err != nil && !errors.IsNotFound(err) {
+			logrus.Errorf("Get node config with nodeName %s failure: %v", nodeName, err)
+			return err
+		}
+		logrus.Debugf("Nodeconfig content: %+v, notFound: %v", nodeConfig, errors.IsNotFound(err))
+		if nodeConfig == nil || errors.IsNotFound(err) {
+			nodeConfig := generateNodeConfig(nodeName, ntpServers)
+			_, err := h.nodeConfigs.Create(nodeConfig)
+			if err != nil {
+				logrus.Errorf("Create node config with nodeName %s failure: %v", nodeName, err)
+			}
```

Comment:
Looks each NODE will have a related nodeconfig, but `ntp-server` should be a global config, each node should use the same (set) of NTP servers, do we allow config ntp server separately?

Or is it only used to reflect that status of applying in each NODE ?

Reply:
the harvester installer already allow configuration of a NTP server on a per node basis. Would this change mean that a common NTP server is rolled out to all nodes in the cluster overwriting the one specified during the install?

Reply:
>Looks each NODE will have a related nodeconfig, but ntp-server should be a global config, each node should use the same (set) of NTP servers, do we allow config ntp server separately?
>
>Or is it only used to reflect that status of applying in each NODE ?

The node settings could be created as cluster or node levels. Currently, the NTP setting is only configurable on the cluster level. Not only NTP but other system-level settings could be set up by the node-manager. We keep the capability on node-manager to let it work on the node-level.

So that is why the `ntp-servers` is now cluster level, but we will create it like node-level.

>the harvester installer already allow configuration of a NTP server on a per node basis. Would this change mean that a common NTP server is rolled out to all nodes in the cluster overwriting the one specified during the install?

Yes, that will be overwritten when we set `ntp-server` on UI.

The user will set the NTP server during the installation. After that, he may want to change the `ntp-server`, it will overwrite the previous values.

Reply:
After some discussion with @Vicente-Cheng , I'd suggest we adjust the setting to this format (a JSON string):
```json
{
  "ntpServers": [
    "server1",
    "server2"
  ],
  "override": {
    "node1": {
      "ntpServers": [
        "server3",
        "server4"
      ]
    },
    "node2": {
      "ntpServers": [
        "server5"
      ]
    }
  }
}
```

The `override` filed contains a map of node-specific settings. By default, the global `ntpServers`setting is applied on each node unless there is a user-specified overriding.

Reply:
After discussion, we do not support different NTP configs on each node. Instead, we use the global NTP config, and if the nodes cross region, we could set multiple NTP servers to make it works.